mem/interface.md

```diff
@@ -132,3 +132,20 @@ This is a very basic memory interface that is incomplete in at least the followi
 * Maybe we want operations that can compare pointers without casting them to integers. Or else we decide only the address can matter for comparison.
 
 [`Ordering`]: https://doc.rust-lang.org/nightly/core/sync/atomic/enum.Ordering.html
+
+```rust
+impl<Provenance> Pointer<Provenance> {
+    /// Offsets the `self` pointer.
+    /// Returns `Ǹone` if the resulting pointer does not fit into `PTR_SIZE`.
+    /// This does not check whether the offset stays in bounds of it's allocation.
+    pub fn offset<M: Memory<Provenance=Provenance>>(self, offset: Size) -> Option<Self> {
```

Comment:
It's a bit strange that this is only used for globals... I think you can make this `wrapping_offset` and change the return type to just `Self` and then use it for ptr_offset_wrapping ?